docs/dev/plugin-quickstart.rst

```diff
@@ -0,0 +1,914 @@
+
+In most cases, we hope that the components defined in HydroMT `Model` classes (``config``, ``geoms``, ``maps``, ``forcing``, ``states``,
+``results``) and its generic subclasses (``grid``, ``mesh``, ``response_units``) should allow you to store any data required by your
+model in a proper way. If it is not the case, you can always define your own new model components by respecting the following steps
+(example if your model has a lot of 2D non-geospatial tabular data that could nicely be stored as pandas.DataFrame objects, *tables*):
+
+1. Initiliase your new component placeholder in the ``__init__`` function, if possible with None.
+
+.. code-block:: python
+
+  self._tables = None
+
+2. Define the component itself as a new property, that looks for the placeholder and tries reading if empty in read mode.
+
+.. code-block:: python
+
+  @property
+  def tables(self) -> Dict[str, pd.DataFrame]:
+      """Returns a dictionary of pandas.DataFrame tabular files."""
+      if self._tables is None:
+        self._tables = dict()
+        if self._read:
+          self.read_tables()
+      return self._tables
+
+3. Define a reading and a writting method for your new component.
+
+.. code-block:: python
+
+  def read_tables(self, **kwargs):
+    """Read table files at <root> and parse to dict of dataframes"""
+    if not self._write:
+      self._tables = dict()  # start fresh in read-only mode
+
+    self.logger.info("Reading model table files.")
+    fns = glob.glob(join(self.root, f"*.csv"))
+    if len(fns) > 0:
+      for fn in fns:
+        name = basename(fn).split(".")[0]
+        tbl = pd.read_csv(fn)
+        self.set_tables(tbl, name=name)
+
+  def write_tables(self):
+    """Write tables at <root>."""
+    self._assert_write_mode
+    if len(self.tables) > 0:
+      self.logger.info("Writing table files.")
+      for name in self.tables:
+        fn_out = join(self.root, f"{name}.csv")
+        self.tables[name].to_csv(fn_out, sep=",", index=False, header=True)
+
+4. Define a set function in order to add or update data in your component.
+
+.. code-block:: python
+
+  def set_tables(self, df: pd.DataFrame, name:str):
+    """Add table <pandas.DataFrame> to model."""
+    if not (isinstance(df, pd.DataFrame) or isinstance(df, pd.Series)):
+      raise ValueError("df type not recognized, should be pandas.DataFrame or pandas.Series.")
+    if name in self._tables:
+      if not self._write:
+        raise IOError(f"Cannot overwrite table {name} in read-only mode")
+      elif self._read:
+        self.logger.warning(f"Overwriting table: {name}")
+    self._tables[name] = df
+
+.. _plugin_workflows:
+
+Data processing: setup methods and workflows
+--------------------------------------------
+Now let's focus on the wheel part of the schematic diagram or how to add data to your model using HydroMT, which is what it's all about!
+Nothing here is really mandatory but more about conventions or tips also from experience with other plugins.
```

Comment:
I think the key message here should be the three steps from you diagram that are required in each setup method, i.e., to read with the data catalog, process with workflows, set/update model properties

Reply:
I now see this is mentioned in the next section. Perhaps change the order?

Reply:
I think this information is also present in other sections and can be improved when we go for the next iteration on the diagrams so I've removed it for now.